hpack/table.py

```diff
@@ -172,12 +185,15 @@ def search(self, name, value):
         """
         offset = HeaderTable.STATIC_TABLE_LENGTH + 1
         partial = None
-        for (i, (n, v)) in enumerate(HeaderTable.STATIC_TABLE):
-            if n == name:
-                if v == value:
-                    return i + 1, n, v
-                elif partial is None:
-                    partial = (i + 1, n, None)
+
+        header_name_search_result = HeaderTable.STATIC_TABLE_MAPPING.get(name)
+        if header_name_search_result is not None:
+            index = header_name_search_result.positions.get(value)
+            if index is not None:
+                return index + 1, name, value
+            else:
+                partial = (header_name_search_result.min + 1, name, None)
```

Comment:
Dynamic entries would be more difficult to simulate "pushing" down the table but probably could be done. 2 hash lookups plus whatever is necessary to track updates compared to iterating over a header table, unpacking tuples, and checking values. Might be something to think about?

I was just suggesting using this method for the static table where you don't have to track changes.